File: frontend/src/types/task.ts
export type TaskStatus =
  | "pending"
  | "completed";

export interface Task {
  id: number;
  user_id: number;
  title: string;
  description: string | null;
  status: TaskStatus;
  due_date: string | null;
}

export interface TaskCreate {
  title: string;
  description: string | null;
  status: TaskStatus;
  due_date: string | null;
}

File: backend/app/schemas/task.py
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class TaskStatus(str, Enum):
    PENDING = "pending"
    COMPLETED = "completed"


class TaskBase(BaseModel):
    title: str
    description: str | None = None
    status: TaskStatus = TaskStatus.PENDING
    due_date: datetime | None = None


class TaskCreate(TaskBase):
    pass


class TaskResponse(TaskBase):
    id: int
    user_id: int

    class Config:
        from_attributes = True